server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2101,9 +2107,10 @@ protected InputStream openSlice(int slice) throws IOException {
                                     return container.readBlob(fileInfo.partName(slice));
```

Comment:
Maybe put `ensureNotClosing` here as well? Then you can run your test with random chunk_size as well I think?

Reply:
Sure, makes sense. Not sure why I missed it